tests/libwait/wait.go

```diff
@@ -104,27 +165,71 @@ func waitForVMIPhase(ctx context.Context, phases []v1.VirtualMachineInstancePhas
 	return vmi
 }
 
-func WaitForSuccessfulVMIStartIgnoreWarnings(vmi runtime.Object) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	wp := watcher.WarningsPolicy{FailOnWarnings: false}
-	return WaitForVMIStart(ctx, vmi, 180, wp)
+// WaitForVMIStartOrFailed blocks until the specified VirtualMachineInstance reaches either Failed or Running states
+func WaitForVMIStartOrFailed(vmi *v1.VirtualMachineInstance, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
+	return WaitForVMIPhase(vmi,
+		[]v1.VirtualMachineInstancePhase{v1.Running, v1.Failed},
+		WithWarningsPolicy(&wp),
+		WithTimeout(seconds),
+		WithWaitForFail(true),
+	)
 }
 
-func WaitForSuccessfulVMIStartWithTimeout(vmi runtime.Object, seconds int) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	wp := watcher.WarningsPolicy{FailOnWarnings: true}
-	return WaitForVMIStart(ctx, vmi, seconds, wp)
+// WaitForVMIScheduling blocks until the specified VirtualMachineInstance scheduled and return the target node name.
+func WaitForVMIScheduling(vmi *v1.VirtualMachineInstance, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
+	return WaitForVMIPhase(vmi,
+		[]v1.VirtualMachineInstancePhase{v1.Scheduling, v1.Scheduled, v1.Running},
+		WithWarningsPolicy(&wp),
+		WithTimeout(seconds),
+	)
 }
 
-func WaitForSuccessfulVMIStartWithTimeoutIgnoreWarnings(vmi runtime.Object, seconds int) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	wp := watcher.WarningsPolicy{FailOnWarnings: false}
-	return WaitForVMIStart(ctx, vmi, seconds, wp)
+// WaitForSuccessfulVMIStart blocks until the specified VirtualMachineInstance reaches the Running state
+func WaitForSuccessfulVMIStart(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+	return WaitForVMIPhase(vmi,
+		[]v1.VirtualMachineInstancePhase{v1.Running},
+	)
 }
 
+// WaitForSuccessfulVMIStartIgnoreWarnings blocks until the specified VirtualMachineInstance reaches the Running state ignoring the warnings
```

Comment:
nit: do we really need all those wrappers, i.e. isn't the refactored code already readable enough?

Maybe we should either inline these or put a deprecation flag on them?

Reply:
A follow up PR about this would be very nice.

Reply:
I agree with both :)